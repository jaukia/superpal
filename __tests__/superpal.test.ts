import superpal from '../src/index';

test("superpal (no args)", () => {
  expect(superpal("#60a5fa")).toMatchSnapshot();
});

test("superpal many args (hsl)", () => {
  expect(superpal("#60a5fa", {colorSpace:"hsl"})).toMatchSnapshot();
});

test("superpal many args (hsluv)", () => {
  expect(superpal("#60a5fa", {colorSpace:"hsluv"})).toMatchSnapshot();
});