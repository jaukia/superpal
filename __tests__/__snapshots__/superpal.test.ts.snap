// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`superpal (no args) 1`] = `
Object {
  "blue": Object {
    "0": "#edf9ff",
    "100": "#d8effe",
    "200": "#bddffe",
    "300": "#9ecbfe",
    "400": "#6aa8fc",
    "500": "#5494fa",
    "600": "#3a7ef4",
    "700": "#1c64dd",
    "800": "#1c51ac",
    "900": "#163770",
  },
  "cyan": Object {
    "0": "#e0fefc",
    "100": "#b0fcf9",
    "200": "#59f5f5",
    "300": "#28e1e7",
    "400": "#18bbce",
    "500": "#15a7c7",
    "600": "#1091b5",
    "700": "#0b7897",
    "800": "#0f6078",
    "900": "#0e4050",
  },
  "fuchsia": Object {
    "0": "#fdf3fb",
    "100": "#fbe2f6",
    "200": "#f8c9f0",
    "300": "#f6a7ee",
    "400": "#e36ef5",
    "500": "#c361f9",
    "600": "#aa47f7",
    "700": "#9321de",
    "800": "#7420af",
    "900": "#4c1d72",
  },
  "gray": Object {
    "0": "#f5f7f7",
    "100": "#e9ebed",
    "200": "#d7dbde",
    "300": "#c2c7cd",
    "400": "#a0a7af",
    "500": "#9197a0",
    "600": "#7f848e",
    "700": "#686d76",
    "800": "#53575e",
    "900": "#373a3f",
  },
  "green": Object {
    "0": "#e9fee0",
    "100": "#c8fdb1",
    "200": "#8afa64",
    "300": "#50eb4a",
    "400": "#40c44e",
    "500": "#39b14c",
    "600": "#319c45",
    "700": "#288138",
    "800": "#296632",
    "900": "#1e4422",
  },
  "indigo": Object {
    "0": "#f8f4ff",
    "100": "#ece7fe",
    "200": "#d7d5fe",
    "300": "#bdc1fe",
    "400": "#929cfd",
    "500": "#7d89fc",
    "600": "#6771f9",
    "700": "#514ff0",
    "800": "#3f3dc3",
    "900": "#2a2d7d",
  },
  "lime": Object {
    "0": "#edfed9",
    "100": "#d2fc99",
    "200": "#acf442",
    "300": "#8ee22b",
    "400": "#69c02d",
    "500": "#54af31",
    "600": "#449b2f",
    "700": "#388026",
    "800": "#326527",
    "900": "#23431c",
  },
  "metadata": Object {
    "analogous30": Array [
      "indigo",
      "cyan",
    ],
    "analogous60": Array [
      "violet",
      "teal",
    ],
    "complementary": "orange",
    "input": "#60a5fa",
    "main": "blue",
  },
  "orange": Object {
    "0": "#fff5e7",
    "100": "#ffe8c7",
    "200": "#fed296",
    "300": "#fdb75d",
    "400": "#e88e1c",
    "500": "#db7915",
    "600": "#c6660e",
    "700": "#a55308",
    "800": "#83430c",
    "900": "#582d0c",
  },
  "pink": Object {
    "0": "#fff3f7",
    "100": "#fee3ed",
    "200": "#fdc9dd",
    "300": "#feaaca",
    "400": "#fc6ea1",
    "500": "#fa4a82",
    "600": "#eb2466",
    "700": "#c61453",
    "800": "#9d1742",
    "900": "#68162d",
  },
  "red": Object {
    "0": "#fff3f3",
    "100": "#fee4e4",
    "200": "#fecccb",
    "300": "#fdafa9",
    "400": "#fb786e",
    "500": "#f7594f",
    "600": "#ea3530",
    "700": "#c52523",
    "800": "#992722",
    "900": "#651f1a",
  },
  "teal": Object {
    "0": "#e2ffef",
    "100": "#b5fed8",
    "200": "#58fcb6",
    "300": "#2de9a0",
    "400": "#27c382",
    "500": "#26b171",
    "600": "#239b61",
    "700": "#1c814f",
    "800": "#1f6640",
    "900": "#18442b",
  },
  "violet": Object {
    "0": "#fbf4fe",
    "100": "#f5e4fd",
    "200": "#eacdfd",
    "300": "#dab4fd",
    "400": "#ba89fd",
    "500": "#a178fc",
    "600": "#8b5efa",
    "700": "#7932f1",
    "800": "#602bbf",
    "900": "#3f237b",
  },
  "yellow": Object {
    "0": "#f8fbd0",
    "100": "#f0f47a",
    "200": "#e0e434",
    "300": "#cbd119",
    "400": "#b6aa13",
    "500": "#b29311",
    "600": "#a77c0e",
    "700": "#946108",
    "800": "#7b490c",
    "900": "#562f0c",
  },
}
`;

exports[`superpal many args (hsl) 1`] = `
Object {
  "blue": Object {
    "0": "#f0fcff",
    "100": "#dcf5fe",
    "200": "#bfe3fd",
    "300": "#9bcbfd",
    "400": "#5fa0fc",
    "500": "#4186fb",
    "600": "#1e6afa",
    "700": "#0553e6",
    "800": "#0a45b2",
    "900": "#0d3072",
  },
  "cyan": Object {
    "0": "#f0fffe",
    "100": "#dcfefc",
    "200": "#bffdfb",
    "300": "#9bfafd",
    "400": "#5fdffc",
    "500": "#41b9fb",
    "600": "#1e97fa",
    "700": "#0581e6",
    "800": "#0a67b2",
    "900": "#0d4572",
  },
  "fuchsia": Object {
    "0": "#fff0fd",
    "100": "#fedcfb",
    "200": "#fdbff8",
    "300": "#fd9bf7",
    "400": "#ef5ffc",
    "500": "#cc41fb",
    "600": "#ae1efa",
    "700": "#9705e6",
    "800": "#780ab2",
    "900": "#4f0d72",
  },
  "gray": Object {
    "0": "#f7f8f8",
    "100": "#eceeef",
    "200": "#dbdee1",
    "300": "#c8ccd0",
    "400": "#a7acb4",
    "500": "#969ca6",
    "600": "#838996",
    "700": "#6c727f",
    "800": "#575c66",
    "900": "#3b3e44",
  },
  "green": Object {
    "0": "#f1fff0",
    "100": "#dffedc",
    "200": "#c1fdbf",
    "300": "#9bfd9d",
    "400": "#5ffc69",
    "500": "#41fb50",
    "600": "#1efa32",
    "700": "#05e619",
    "800": "#0ab219",
    "900": "#0d7216",
  },
  "indigo": Object {
    "0": "#f6f0ff",
    "100": "#e7dcfe",
    "200": "#cbbffd",
    "300": "#a49bfd",
    "400": "#645ffc",
    "500": "#4241fb",
    "600": "#1e21fa",
    "700": "#0507e6",
    "800": "#0a0cb2",
    "900": "#0d0f72",
  },
  "lime": Object {
    "0": "#f8fff0",
    "100": "#effedc",
    "200": "#dffdbf",
    "300": "#c9fd9b",
    "400": "#a0fc5f",
    "500": "#85fb41",
    "600": "#69fa1e",
    "700": "#51e605",
    "800": "#43b20a",
    "900": "#30720d",
  },
  "metadata": Object {
    "analogous30": Array [
      "indigo",
      "cyan",
    ],
    "analogous60": Array [
      "violet",
      "teal",
    ],
    "complementary": "orange",
    "input": "#60a5fa",
    "main": "blue",
  },
  "orange": Object {
    "0": "#fffaf0",
    "100": "#fef3dc",
    "200": "#fde6bf",
    "300": "#fdd39b",
    "400": "#fca85f",
    "500": "#fb8241",
    "600": "#fa5d1e",
    "700": "#e64505",
    "800": "#b23a0a",
    "900": "#722a0d",
  },
  "pink": Object {
    "0": "#fff0f7",
    "100": "#fedced",
    "200": "#fdbfdc",
    "300": "#fd9bc8",
    "400": "#fc5f9a",
    "500": "#fb4174",
    "600": "#fa1e4f",
    "700": "#e60536",
    "800": "#b20a2f",
    "900": "#720d24",
  },
  "red": Object {
    "0": "#fff0f1",
    "100": "#fedcdf",
    "200": "#fdbfc2",
    "300": "#fd9d9b",
    "400": "#fc685f",
    "500": "#fb4e41",
    "600": "#fa2f1e",
    "700": "#e61605",
    "800": "#b2170a",
    "900": "#72150d",
  },
  "teal": Object {
    "0": "#f0fff8",
    "100": "#dcfeef",
    "200": "#bffde2",
    "300": "#9bfdd1",
    "400": "#5ffcb2",
    "500": "#41fb9d",
    "600": "#1efa88",
    "700": "#05e671",
    "800": "#0ab25b",
    "900": "#0d723e",
  },
  "violet": Object {
    "0": "#faf0ff",
    "100": "#f3dcfe",
    "200": "#e7bffd",
    "300": "#d39bfd",
    "400": "#a95ffc",
    "500": "#8541fb",
    "600": "#621efa",
    "700": "#4a05e6",
    "800": "#3e0ab2",
    "900": "#2c0d72",
  },
  "yellow": Object {
    "0": "#fefff0",
    "100": "#fdfedc",
    "200": "#fbfdbf",
    "300": "#f8fd9b",
    "400": "#fcef5f",
    "500": "#fbcb41",
    "600": "#fa9c1e",
    "700": "#e65e05",
    "800": "#b2370a",
    "900": "#721c0d",
  },
}
`;

exports[`superpal many args (hsluv) 1`] = `
Object {
  "blue": Object {
    "0": "#eef8ff",
    "100": "#daeefe",
    "200": "#c0ddfd",
    "300": "#a4c9fd",
    "400": "#6fa7fc",
    "500": "#5795fb",
    "600": "#2980fa",
    "700": "#166ad4",
    "800": "#2055a6",
    "900": "#1d3a6e",
  },
  "cyan": Object {
    "0": "#d4fffe",
    "100": "#86fefd",
    "200": "#38f1f3",
    "300": "#28dbe2",
    "400": "#20b6c6",
    "500": "#1ca4be",
    "600": "#188fad",
    "700": "#127790",
    "800": "#1b5e72",
    "900": "#19404d",
  },
  "fuchsia": Object {
    "0": "#fff3fd",
    "100": "#fee3fa",
    "200": "#fdcaf7",
    "300": "#fdabf7",
    "400": "#f074fc",
    "500": "#d267fb",
    "600": "#bc4dfa",
    "700": "#ab18ea",
    "800": "#8522b6",
    "900": "#581f78",
  },
  "gray": Object {
    "0": "#f6f6f7",
    "100": "#eaebed",
    "200": "#d8dadd",
    "300": "#c4c7cb",
    "400": "#a2a6ae",
    "500": "#92969f",
    "600": "#80848c",
    "700": "#6a6d74",
    "800": "#55575d",
    "900": "#3a3b3f",
  },
  "green": Object {
    "0": "#e2ffdb",
    "100": "#b3fe9d",
    "200": "#69f733",
    "300": "#30e523",
    "400": "#1cbf35",
    "500": "#18ad39",
    "600": "#149834",
    "700": "#0f7e2a",
    "800": "#176526",
    "900": "#16441d",
  },
  "indigo": Object {
    "0": "#faf4ff",
    "100": "#f1e7fe",
    "200": "#ded4fd",
    "300": "#c6c0fd",
    "400": "#9a9dfc",
    "500": "#7e8dfb",
    "600": "#5c79fa",
    "700": "#265ef9",
    "800": "#244cc5",
    "900": "#203581",
  },
  "lime": Object {
    "0": "#e5ffcf",
    "100": "#befe6a",
    "200": "#a3ef33",
    "300": "#89dc23",
    "400": "#66ba1c",
    "500": "#50aa18",
    "600": "#3f9614",
    "700": "#337c0f",
    "800": "#2c6317",
    "900": "#204316",
  },
  "metadata": Object {
    "analogous30": Array [
      "indigo",
      "cyan",
    ],
    "analogous60": Array [
      "violet",
      "teal",
    ],
    "complementary": "orange",
    "input": "#60a5fa",
    "main": "blue",
  },
  "orange": Object {
    "0": "#fff5eb",
    "100": "#fee7cf",
    "200": "#fdd3a2",
    "300": "#fdb960",
    "400": "#e8921c",
    "500": "#df7d19",
    "600": "#cc6914",
    "700": "#aa560f",
    "800": "#874618",
    "900": "#5b3016",
  },
  "pink": Object {
    "0": "#fff4f7",
    "100": "#fee5ed",
    "200": "#fdcdde",
    "300": "#fdb1cd",
    "400": "#fc7ca9",
    "500": "#fc5d8c",
    "600": "#fc236b",
    "700": "#d51157",
    "800": "#a81947",
    "900": "#721831",
  },
  "red": Object {
    "0": "#fff4f5",
    "100": "#fee5e7",
    "200": "#fdcfd1",
    "300": "#fdb4b3",
    "400": "#fd8179",
    "500": "#fc6452",
    "600": "#f24215",
    "700": "#ca360f",
    "800": "#a02e18",
    "900": "#6c2216",
  },
  "teal": Object {
    "0": "#dcffeb",
    "100": "#a1fecf",
    "200": "#35f7ac",
    "300": "#25e19a",
    "400": "#1dbd7d",
    "500": "#19ab6d",
    "600": "#15975d",
    "700": "#0f7d4c",
    "800": "#18633e",
    "900": "#17432c",
  },
  "violet": Object {
    "0": "#faf4ff",
    "100": "#f4e6fe",
    "200": "#ead1fd",
    "300": "#dbb9fd",
    "400": "#bb92fc",
    "500": "#9e83fb",
    "600": "#826ffa",
    "700": "#6a4df9",
    "800": "#542de7",
    "900": "#392498",
  },
  "yellow": Object {
    "0": "#faff61",
    "100": "#eff338",
    "200": "#dde233",
    "300": "#c8ce24",
    "400": "#b2a91c",
    "500": "#ad9518",
    "600": "#a37e14",
    "700": "#92640f",
    "800": "#7b4d17",
    "900": "#583216",
  },
}
`;
